cmd/sriovdp/manager.go

```diff
@@ -290,3 +279,58 @@ type linkWatcher struct {
 func (lw *linkWatcher) Subscribe() {
 
 }
+
+func (rm *resourceManager) netDevHandler(dev *ghw.PCIDevice) {
+	// exclude device in-use in host
+	if isDefaultRoute, _ := hasDefaultRoute(dev.Address); !isDefaultRoute {
+
+		aPF := utils.IsSriovPF(dev.Address)
+		aVF := utils.IsSriovVF(dev.Address)
+
+		if aPF || !aVF {
+			// add to linkWatchList
+			rm.addToLinkWatchList(dev.Address)
+		}
+
+		if aPF && utils.SriovConfigured(dev.Address) {
+			// do not add this device in net device list
+			return
+		}
+
+		if newDevice, err := resources.NewPciNetDevice(dev, rm.rFactory); err == nil {
+			rm.netDeviceList = append(rm.netDeviceList, newDevice)
+		} else {
+			glog.Errorf("discoverDevices() error adding new device: %q", err)
+		}
+
+	}
+}
+
+func (rm *resourceManager) accDevHandler(dev *ghw.PCIDevice) {
+	//Check if device is a physical function (PF)
+	aPF := utils.IsSriovPF(dev.Address)
+	devID, err := strconv.ParseInt(dev.Product.ID, 16, 64)
+	if err != nil {
+		glog.Warningf("discoverDevices(): unable to parse device ID for device %+v %q", dev, err)
+		return
+	}
+	//Check if device ID = 0b30, it means it is a RSU (Remote System Update) device and not accelerator
+	if devID != rsuDevice {
+		if aPF && utils.SriovConfigured(dev.Address) {
+			// If it is a PF and SRIOV is configured do not add to device list
+			return
+		}
+
+		if newDevice, err := resources.NewPciNetDevice(dev, rm.rFactory); err == nil {
+			if newDevice.GetDriver() == "igb_uio" || newDevice.GetDriver() == "vfio-pci" {
+				rm.netDeviceList = append(rm.netDeviceList, newDevice)
+			} else {
+				glog.Infof("Excluding FEC device %-12s because it is not bound to userspace driver.", dev.Address)
+			}
+		} else {
+			glog.Errorf("discoverDevices() error adding new device: %q", err)
+		}
+	} else {
```

Comment:
It might be better to check if devID == rsuDevice { log and return } at line 318.  I'm thinking of the (future) case of any other accelerator classed PCI devices that would not be appropriate to discover here.  It might be useful to have a list of 'blacklisted' device Ids per class that can be looped through rather than individual checks.